FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenManagerTest.m

```diff
@@ -70,7 +72,10 @@ - (void)setUp {
 }
 
 - (void)tearDown {
-  [_messaging.tokenManager stopAllTokenOperations];
+  [_messaging.tokenManager.tokenOperations addBarrierBlock:^{
```

Comment:
I guess, [waitUntilAllOperationsAreFinished](https://developer.apple.com/documentation/foundation/nsoperationqueue/1407971-waituntilalloperationsarefinishe) method may do the trick here.

Reply:
I tried that one and that one waits indefinitely.

Reply:
`stopAllTokenOperations` and `stopAllTokenOperations` both worked locally. `stopAllTokenOperations` didn't work as i expected it to on CI though.

Reply:
In the case `waitUntilAllOperationsAreFinished` creates a deadlock, you may use `addBarrierBlock` along with an expectation to postpone tearDown completion.

Reply:
If I'm not mistaken, `addBarrierBlock` returns immediately, so you will need an expectation here e.g.:
```
XCTestExpectation *exp = [...];
[_messaging.tokenManager.tokenOperations addBarrierBlock:^{
  [exp fulfill];
}];

[self waitForExpectations: @[exp]];
```